### PR TITLE
feat(framework): make experiments reproducible via seeding and deterministic execution

### PR DESCRIPTION
**Pull request by amadeuszsz**
## Summary

- Seed all random number generators (Python, NumPy, torch, CUDA) through `seed_everything(seed, workers=True)` with a global `seed` defined in the trainer defaults, so every run — including dataloader workers — starts from a reproducible state.
- Enable `deterministic: warn` on the trainer: torch ops take deterministic kernel paths, and any future op without a deterministic implementation announces itself in the logs instead of silently diverging.
- Derive `PipelineContext`'s RNG from the globally seeded NumPy state instead of an unseeded generator, making secondary-sample draws in transforms (e.g. mix-style augmentations) reproducible per worker.
- Export `CUBLAS_WORKSPACE_CONFIG=:4096:8` via the pixi activation environment, so deterministic cuBLAS is guaranteed for both container and local pixi installations from a single source (requires an image rebuild to reach baked containers).
- Hardcode `deterministic=True` in PTv3's flash-attention call: the default backward accumulates `dq` with atomic adds, producing non-reproducible gradients; the deterministic variant costs ≤12% on the attention op (low single-digit % per training step) and does not affect inference or export.
- Pin the RNG explicitly in the `FrustumMix` test to keep it independent of global seeding state.

### Validation

- Dataloader batches, model initialization, and the full first device-side batch are bit-identical across independent processes (SHA-256 verified); a different seed produces different digests.
- Two 10-step PTv3 training runs reproduced bit-exactly (identical per-step losses and final-weights hash) across separate processes, modulo the spconv limitation below.

### Known limitation

- spconv (2.3.6) selects kernel configurations by benchmarking them at first call; this once-per-process timing-based pick chooses between a small set of near-tied kernels, each individually deterministic (~1e-3 relative loss spread, no metric impact). It affects forward passes too, so bit-exact equality across runs is not guaranteed until spconv offers a deterministic selection policy — there is no supported knob in 2.3.6.

## Checklist

- [ ] You've checked our [contribution overview](https://tier4.github.io/autoware-ml/contributing/contribution-overview/).
- [ ] Your PR follows our [pull request guidelines](https://tier4.github.io/autoware-ml/contributing/pull-request-guidelines/).

### CODE REVIEW

File: autoware_ml/models/segmentation3d/encoders/ptv3.py [hunk 1]
```diff
@@ -518,6 +518,7 @@ def forward(self, point: Point) -> Point:
                 max_seqlen=patch_size,
                 dropout_p=self.attn_drop_p if self.training else 0.0,
                 softmax_scale=self.scale,
+                deterministic=True,
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/models/segmentation3d/encoders/ptv3.py:521]: https://github.com/tier4/autoware-ml/pull/75/commits/abe7e2900cfb64dee10b8cbb2c5aaf24f45175f3
